### ISSUE TITLE
scale_x_discrete fails in combination with geom_histogram

### ISSUE THREAD

**Issue by lzamparo** [has2k1/plotnine#70]
Hi,

I just found plotnine yesterday, and am keen to use it, but I ran into an issue with custom scales. 
 I'm trying to use `scale_x_discrete` to make a more informative x-axis for my histogram, but the scale does not render properly, all the ticks and marks are clustered around the origin.  Here's the gist of how I make the histogram:  https://gist.github.com/lzamparo/5419bf2278897e6e9ea6d70770be1532

Here's the default scale histogram:

![rebuilt_atlas_length_histogram_vanilla](https://user-images.githubusercontent.com/1569186/31827367-8fe74362-b585-11e7-81ba-102e06564a87.png)

The scale is simple enough to create:

```
limits = (0,1500)
major_breaks = breaks.mpl_breaks()(limits)
labels = [str(l) for l in major_breaks]
major_breaks
array([    0.,   150.,   300.,   450.,   600.,   750.,   900.,  1050.,
        1200.,  1350.,  1500.])
labels
['0.0', '150.0', '300.0', '450.0', '600.0', '750.0', '900.0', '1050.0', '1200.0', '1350.0', '1500.0']
```

But the resulting figure collapses the scale onto the origin:

`gg.scales.scale_x_discrete(breaks = major_breaks, labels = labels, limits=limits)`

![rebuilt_atlas_length_histogram](https://user-images.githubusercontent.com/1569186/31827120-000387d8-b585-11e7-9c86-5df8881d1937.png)

Any idea what's gone wrong?

**Comment by has2k1** [comment]
1. You are using continuous data, you should be using a continuous scale. 

2. Do not to compute (`major_breaks = breaks.mpl_breaks()(limits)`) the breaks directly. If you want to you a different breaks computing algorithm, supply the function.

```python
+ gg.scales.scale_x_continuous(breaks=breaks.extended_breaks(), limits=limits)
```

**Comment by lzamparo** [comment]
Thanks for clearing up how to properly use breaks

**Comment by lzamparo** [comment]
If possible, could the documentation for specifying a [scale](https://plotnine.readthedocs.io/en/stable/generated/plotnine.scales.scale.scale.html#plotnine.scales.scale.scale) specify that `breaks` expects a function which returns `array_like` based on `limits`, rather than an `array_like`?  I was passing an `array_like`, and it didn't work.  I'm happy to submit a PR for the docs.

Alternatively, I'm happy to put together an addition to the gallery which details the difference.  Which do you prefer?

**Comment by has2k1** [comment]
You were not the [first](https://github.com/has2k1/plotnine/issues/34) to run against the insufficient documentation. It was fixed in #35, results [here]( http://plotnine.readthedocs.io/en/dev/generated/plotnine.scales.scale.scale.html)

For the gallery if you have anything that shows off any cool/unusual graph feel free to submit an [example](https://github.com/has2k1/plotnine-examples). There is also a slowly growing [tutorial](http://plotnine.readthedocs.io/en/dev/tutorials.html#miscellaneous) section for the "niggly" little things.

**Comment by lzamparo** [comment]
Will keep this in mind.  The example I put up for this issue isn't very interesting, but probably something in the course of this project will be worth submitting.  Thanks again for providing this great framework!
